sds_in_a_box/SDSCode/opensearch_utils/query.py

```diff
@@ -0,0 +1,101 @@
+import json
+
+class Query:
+    """
+    Query class to represent an AWS OpenSearch query domain-specific language (DSL),
+    providing a range of search options. Query objects will be used by the Client
+    search method and used in OpenSearch searches.
+
+    ...
+
+    Attributes
+    ----------
+    query: dict
+        dictionary containing the query field (dict key) and the query text 
+        (dict value).
+    size: int, optional
+        number of results to return. default is 10.
+    
+
+    Methods
+    -------
+    query_dsl: str
+        returns the query in the OpenSearch Query DSL format
+    size: int
+        returns the number of results the query is allowed to return in the search.
+    """
+
+    def __init__(self, query_params, size=10):
+        self.query_params = query_params
+        self.query_dsl_formatted = self.__build_query_dsl(query_params)
+        self.query_size = size 
+
+    def query_dsl(self):
+        """Returns the query parameters in the AWS OpenSearch Query DSL format"""
+        return json.dumps(self.query_dsl_formatted)
+
+    def size(self):
+        """Returns the number of results the query is allowed to return in the search"""
+        return self.query_size
+
+    def __build_query_dsl(self, query_params):
+        """
+        Builds a Query DSL using a dictionary with field:value pairings.
+
+        Parameters
+        ----------
+        query_params: dict
+            dictionary containing field:value search parameters.
+
+        """
+        # define the query structure
+        query = {"query": {"bool":{}}}
+        query_match_structure = {"match": {}}
+        query_date_structure = {"range": {"date": {}}}
+        
+        # remove all params that are not valid
+        query_params = self.__filter_params(query_params)
+
+        # create the query
+        for param in query_params:
+            # create a date query using start and end date parameters
+            if param == "start_date" or param == "end_date":
```

Comment:
I remember things being tricky when you query opensearch. In this function, you are trying to construct this query logic
```
{
  "query": {
    "bool": {
      "filter": {
        "range": {
          "date": {
            "gte": start_date,
            "lte": end_date
          }
        }
      }
    }
  }
}
```
is that right?

Reply:
That's correct